File: backend/src/main/java/com/app/backend/dao/entities/Horaire.java
package com.app.backend.dao.entities;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.sql.Time;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.springframework.data.mongodb.core.mapping.Document;


@Document
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
@Builder
public class Horaire {
    @Id
    private String id;
    private String jour;
    private Time debutMatin;
    private Time finMatin;
    private Time debutApMidi;
    private Time finApMidi;
    @DBRef
    private User medecin;

}

File: frontend/src/app/shared/Horaire/horaire.ts
import { Time } from "@angular/common";
import { User } from "../user";

export class Horaire {
  id: string;
  jour: string;
  debutMatin: Time;
  finMatin: Time;
  debutApMidi: Time;
  finApMidi: Time;
  medecin: User;

  constructor(
    id: string,
    jour: string,
    debutMatin: Time,
    finMatin: Time,
    debutApMidi: Time,
    finApMidi: Time,
    medecin: User
  ) {
    this.id = id;
    this.jour = jour;
    this.debutMatin = debutMatin;
    this.finMatin = finMatin;
    this.debutApMidi = debutApMidi;
    this.finApMidi = finApMidi;
    this.medecin = medecin;
  }
}
